File: L1/PRG_L1_PY/08 [2] 가운데 글자 가져오기.py
# def solution(s):
# 	# 왼쪽이 짝수, 오른쪽이 홀수
# 	answer =  s[int(len(s)/2)-1] + s[int(len(s)/2)]
# 	# 홀수일 때 통과하겠다.
# 	# 2로 나눈 나머지가 1일 때 통과하겠다
# 	if (len(s)%2):
# 		answer = answer[1]
# 	return answer

# 나보다 잘한사람 풀이 있음

def string_middle(str):
    return str[(len(str)-1)//2:len(str)//2+1]
print(string_middle("abcdef"))

File: L1/PRG_L1_C/08 [2] 가운데 글자 가져오기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(const char* s) {
	int i = 0;
	while (*(s + ++i) != NULL) {
	}
	int a = (int)(((float)i - 1) / 2);
	int b = (int)((float)i / 2);
	char* answer = (char*)malloc(sizeof(char)*(b-a));
	for (i = 0; i <= b - a; i++) {
		*(answer + i) = *(s + i + a);
	}
	*(answer + i) = NULL;
	return answer;
}

int main() {
	printf("%s", solution("qwer"));
}
